File: Basic Math/Square root of a number/SquareRoot.java
import java.util.Scanner;

class Solution {
  long floorSqrt(long n) {
    long low = 1, high = n, mid;
    while (low <= high) {
      mid = low + (high - low) / 2;
      if (mid * mid <= n) low = mid + 1;
      else high = mid - 1;
    }
    return high;
  }
}

class SquartRoot {
  public static void main(String args[]) {
    Scanner sc = new Scanner(System.in);
    int t = sc.nextInt();
    while (t > 0) {
      long a = sc.nextInt();
      Solution obj = new Solution();
      System.out.println(obj.floorSqrt(a));
      t--;
      System.out.println("~");
    }
    sc.close();
  }
}

File: Basic Math/Square root of a number/SquareRoot.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
 public:
  long long int floorSqrt(long long int n) {
    long long int low = 1, high = n, mid;
    while (low <= high) {
      mid = low + (high - low) / 2;
      if (mid * mid <= n)
        low = mid + 1;
      else
        high = mid - 1;
    }
    return high;
  }
};

int main() {
  int t;
  cin >> t;
  while (t--) {
    long long n;
    cin >> n;
    Solution obj;
    cout << obj.floorSqrt(n) << endl;

    cout << "~"
         << "\n";
  }
  return 0;
}